src/api/app/controllers/staging/staging_projects_controller.rb

```diff
@@ -34,7 +36,17 @@ def create
   end
 
   def copy
-    authorize @main_project.staging
+    @project = Project.where(name: params[:staging_project_copy_name]).first_or_initialize
```

Comment:
maybe move this ```@project``` assignment to a `before_action`

Reply:
this shouldn't even have a `@` IMO - so I could move the whole authorize into a before filter. but taking that it's only 2 lines more I wonder if it's worth it